src/game/SpellAuras.cpp

```diff
@@ -3912,6 +3912,28 @@ void Aura::HandleInvisibility(bool apply, bool Real)
                     target->SetVisibility(VISIBILITY_ON);
             }
         }
+
+        switch (GetId())
+        {
+        case 38544:
```

Comment:
I'm not sure why is this code required.
In case you want ot unsummon the pet on aura removal, then I think we should make sure that the aura is negative, so that the player won't be able to remove it by click.